src/shared_utilities/DataHandling.jl

```diff
@@ -0,0 +1,246 @@
+module DataHandling
+
+using Dates
+
+import ClimaCore
+
+import ..FileReader: NCFileReader
+import ..Regridder
+import ..Regridder:
+    AbstractRegridder, TempestRegridder, regrid, AVAILABLE_REGRIDDERS
+
+"""
+
+Note: It is best to have one file per variable with all the temporal data.
+
+Assumptions:
+- There is only one file with all the data in time
+- The file has well-defined physical dimensions (e.g., lat/lon)
+- Currently, the time dimension has to be either "time" or "date", the spatial
+  dimensions have to be lat and lon (restriction from TempestRegridder)
+
+DataHandler is meant to live on the CPU, but the Fields can be on the GPU as well.
+"""
+struct DataHandler{
+    FR <: NCFileReader,
+    REG <: AbstractRegridder,
+    SPACE <: ClimaCore.Spaces.AbstractSpace,
+    REF_DATE <: Dates.DateTime,
+    TSTART <: AbstractFloat,
+    DATES <: AbstractArray{<:Dates.DateTime},
+    DIMS,
+    TIMES <: AbstractArray{<:AbstractFloat},
```

Comment:
We discussed this last time, but`TIMES <: AbstractArray{<:AbstractFloat}` is the same as `AbstractArray{<:AbstractFloat}` in this context, right? 

If so, I dont see that we need to have these types as parametric types.